internal/compiler/program.go

```diff
@@ -321,6 +321,9 @@ func (p *Program) getSemanticDiagnosticsForFile(sourceFile *ast.SourceFile) []*a
 	}
 
 	diags := slices.Clip(sourceFile.BindDiagnostics())
+	if sourceFile.CheckJsDirective != nil && !sourceFile.CheckJsDirective.Enabled {
+		return diags
+	}
```

Comment:
Is this the right place for this? I thought the checker itself checked this option?

Reply:
That's what I thought but it wasn't working for some cases without this.

Reply:
Some of the test failures are due to additional comment parsing at the end of this function (i.e for `@ts-expect-error`), which happens outside the checkers.

Reply:
The thing I'm considering is just that we should be looking to see where `SourceFile.checkJsDirective` was used in the old codebase, then ensure we mirror that here.

Reply:
That is what I did for the rest of the PR. But I still needed this for tests to pass. Not sure whyz

Reply:
Okay I dug in some more here and found the equivalent typescript-side check:

https://github.com/microsoft/TypeScript/blob/0693cc72e6ddb6c7468a14e05888efa2f43ac7b0/src/compiler/program.ts#L3058-L3066

```ts
    function getBindAndCheckDiagnosticsForFileNoCache(
    ): readonly Diagnostic[] {
            if (skipTypeChecking(sourceFile, options, program)) {
                return emptyArray;
            }
```

In the typescript impl, `skipTypeChecking` is reused between program and checker. But on the golang side, `skipTypeChecking` only exists in the `internal/checker` package. I can try moving it into `internal/core`.. might make more sense as `(*core.CompilerOptions).SkipTypeChecking(sourceFile)` ?

Reply:
I think I just fixed this in #763. Anders also just redid the pragma parsing for JSX reasons, so I think your PR will need and update but end up simpler as a result.